### ISSUE TITLE
_drawHatchMark throws exception at some edge case

### ISSUE THREAD

**Issue by angel1st** [Ali-Azmoud/flutter_xlider#25]
Hi there,
I  believe I found out an issue, when using the slider. It is quite an edge case and I cannot isolate it in minimum reproducing sample, however, I debugged it and I will try to explain the case:

The issue appears in case `hatchMark` is set.

In case the slider is on app home page, at some point `_drawHatchMark` is called. Within it however, at lines **461** to **465**, the slider uses `_constraintMaxHeight` and `_constraintMaxWidth` for its internal calculations. 
These two class members however are initialized at lines **605** and **606** within `build` method.

In my edge case, however, `_drawHatchMark` is called from `didUpdateWidget` which as I believe you are familiar is actually called **prior** `build` method.
In summary - when `_drawHatchMark` is called the above two class members are `null`, hence the calculations failed.

I hope the above will give you an idea about the case and how to fix it.

Thanks for the excellent widget and I look forward to hearing from you!

**Updated**
Actually commented out `_drawHatchMark` call from `didUpdateWidget` (line **592**) seems to fix the issue, I would however leave to you to decide if this is the ulitmate fix.

**Comment by angel1st** [comment]
Occasionally now, I have got this exception:
> I/flutter ( 3216): ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═══
I/flutter ( 3216): The following NoSuchMethodError was thrown during a scheduler callback:
I/flutter ( 3216): The method 'findRenderObject' was called on null.
I/flutter ( 3216): Receiver: null
I/flutter ( 3216): Tried calling: findRenderObject()
I/flutter ( 3216): 
I/flutter ( 3216): When the exception was thrown, this was the stack:
I/flutter ( 3216): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
I/flutter ( 3216): #1      _FlutterSliderState._renderBoxInitialization (package:flutter_xlider/flutter_xlider.dart:1489:39)
I/flutter ( 3216): #2      _FlutterSliderState._initialize (package:flutter_xlider/flutter_xlider.dart:432:5)
I/flutter ( 3216): #3      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1016:15)
I/flutter ( 3216): #4      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:966:9)
I/flutter ( 3216): #5      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:784:7)
I/flutter ( 3216): #14     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:382:19)
I/flutter ( 3216): #15     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
I/flutter ( 3216): #16     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
I/flutter ( 3216): (elided 8 frames from package dart:async and package dart:async-patch)
I/flutter ( 3216): ══════════════════════════════════════════════════

**Comment by angel1st** [comment]
@Ali-Azmoud - I have just created [PR](https://github.com/Ali-Azmoud/flutter_xlider/pull/26) hopefully with the fix...

**Comment by Ali-Azmoud** [comment]
Hi, sorry for my late response. I am traveling and don't have access to my laptop.
I will fix this bug asap. thanks for your report.

**Comment by angel1st** [comment]
@Ali-Azmoud -thanks for the feedback.

**Comment by Ali-Azmoud** [comment]
Hi @angel1st, sorry for my late response! it's quite a while...
I couldn't reproduce this issue for myself. but it's better to check for `_constraintMaxHeight` and `_constraintMaxWidth` to be not null inside `didUpdateWidget` to avoid any exception. I will merge your solution but change it to my solution!

**Comment by Ali-Azmoud** [comment]
Hey, I couldn't reproduce this issue but changed the `didUpdateWidget ` method. if you encountered the exception again let me know. thanks. I'm going to close this issue. feel free to reopen it if you face any bug or exception
